sdk/eventhub/Azure.Messaging.EventHubs/src/Core/TransportProducerPool.cs

```diff
@@ -0,0 +1,299 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Core
+{
+    /// <summary>
+    ///   A pool of <see cref="TransportProducer" />.
+    ///   It periodically checks if the producers are active and if not it disposes them.
+    /// </summary>
+    ///
+    internal class TransportProducerPool : IAsyncDisposable
+    {
+        /// <summary>The period in minutes after which <see cref="PerformExpiration"/> is run.</summary>
+        private const int PerformExpirationPeriodAsMinutes = 10;
```

Comment:
nit:  You may want to do this as a `static readonly TimeSpan` to save the need to convert it.   That said, it appears only to be used in the constructor, so not a big deal either way.